### PR TITLE
Migrate ERC721 to Cairo2

### LINKED ISSUE #658: Migrate ERC721 to Cairo 2

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #658 

#### PR Checklist

- [x] Tests
- [ ] Tried the feature on a public network
- [ ] Documentation

### CODE REVIEW

**Comment by ericnordelo** [comment]
This PR will be smaller after merging #664 and #666

**Review by andrew-fleming — COMMENTED** [review]
Excellent PR! I left a few small questions

File: src/openzeppelin/tests/mocks/camel721_mock.cairo
Comment (ericnordelo) [reply src/openzeppelin/tests/mocks/camel721_mock.cairo:49]: I think is easier to rewrite external than to define an interface, and the generic implementation in the mock. Do you think is worth it? We have mocks with only one external method in other modules. Should we write the interface and the generic implementation instead of one external as well?

File: src/openzeppelin/tests/token/test_dual721.cairo
Comment (ericnordelo) [reply src/openzeppelin/tests/token/test_dual721.cairo:158]: Yes :(

File: src/openzeppelin/tests/mocks/camel721_mock.cairo
Comment (andrew-fleming) [reply src/openzeppelin/tests/mocks/camel721_mock.cairo:49]: Oh no, I didn't mean define a new interface, I meant reusing the 721 interface (and ISRC5) and just create an impl of the interface inside the mock (to just use one `#[external]`). Just an idea, maybe it's more trouble than it's worth

Comment (ericnordelo) [reply src/openzeppelin/tests/mocks/camel721_mock.cairo:49]: Oh, I got you, I think that makes sense if the interface already contains all the methods. I will update it accordingly.

Comment (ericnordelo) [reply src/openzeppelin/tests/mocks/camel721_mock.cairo:49]: Actually, this doesn't works well with the CamelOnly pattern I've been using, because we don't have the agnostic functions in that trait, and defining the Camel trait just for the mock seems like an overhead to me. Are you ok with continuing like this in the meantime, and maybe refactor this in another PR in the future if we avoid the CamelOnly pattern?

Comment (andrew-fleming) [reply src/openzeppelin/tests/mocks/camel721_mock.cairo:49]: Ahh right. Yeah, explicitly setting each method as external is totally fine by me. It could be a while before we can have duplicate methods

Comment (andrew-fleming) [reply src/openzeppelin/tests/mocks/camel721_mock.cairo:49]: > Or maybe we can get rid of the agnostic methods in the camel mock? I think we are already doing this in CamelERC20Panic

Your call on this. I thought this would make the mocks a little easier 😅

Comment (ericnordelo) [reply src/openzeppelin/tests/mocks/camel721_mock.cairo:49]: I rather let this for the "Improve Mocks" PR if you agree (same for the directory organization of mocks). I created this issue for that #670

**Review by andrew-fleming — APPROVED** [review]
Left a small suggestion, but I think we're good to go! Great work

**Pull request merged**
